Modul_6_Dzhumelya

Uraniadz opened the pull request.

Uraniadz committed:
Modul_6_Dzhumelya

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+
+const arr = ['one', 10, 'name', 5, 'Nastya', 9];
+
+function arrayCopy () {
```
Малося на увазі, що функція повинна приймати масив як вхідний параметр.
```suggestion
function arrayCopy (array) {
```

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+const arr = ['one', 10, 'name', 5, 'Nastya', 9];
+
+function arrayCopy () {
+    return arr.concat();
```
Оригінальний підхід, зараховано :) 
Але краще використати метод `slice()`, так як його пряме призначення - копіювання масиву, і в деяких випадках він може працювати трохи краще (це вже якщо говорити про оптимізацію коду).

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+
+const arrTwo = ['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog'];
+
+function arrayJoin (arr) {
```
Допишіть функцію таким чином, щоб вона приймала розділювач як вхідний параметр
```suggestion
function arrayJoin (arr, delimiter) {
```
`const arrNewJoin = arrayJoin(arrTwo, ' '); `

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+function getWord () {
+    const arr1 = ['e', null, 'r'];
+    const arr2 = ['g', 'y'];
+    const arrUnite = [];
```
А тут вже простіше використати метод `concat`

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+    return arrayJoin(removeFalsy(arrUnite));
+}
+const arrNewUnite = getWord();
+console.log(arrNewUnite);
```
Як опція, для виводу результатів можна обійтися без проміжної константи
```suggestion
console.log(getWord());
```

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+function getWord2 () {
+    const arr1 = ['e', null, 'r'];
+    const arr2 = ['g', 'y'];
+    const arrUnite = [];
```
```suggestion
    const arrUnite = arr1.concat(arr2);
```

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,90 @@
+
+function chunk(array, size) {
+    const newArr = [];
+    if (size === 3) {
```
Пропоную тут пройти циклом по масиву і щоразу, коли позиція елемента не даватиме остачі від ділення на `size` робити push частини масиви в `newArr`

Uraniadz committed:
Corected

Нарешті, зробив вправу 5.
Вправу 6. Так і не получилося розділити масив на 3 частини, навіть через цикл for і операції %

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,79 @@
+    const objectFromArr = {};
+
+    for (let i = 0; i < array.length; i++) {
+        for (let j = 0; j < array[i].length; j++) {
```
Замість цього циклу буде достатньо одного рядка, оскільки за умовою маємо на вході пари елементів
`objectFromArr[array[i][0]] = array[i][1];`

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,79 @@
+    arrUnite.splice(2, 0, 'w');
+    arrUnite.splice(6,2);
+    arrUnite[6] = 'c';
+    return arrayJoin(removeFalsy(arrUnite), ' ');
```
Щоб отримати суцільне слово як розділювач можна використати пустий рядок
```suggestion
    return arrayJoin(removeFalsy(arrUnite), '');
```

vboiko-powercode commented on Modul_6_Dzhumelya/js/index.js:
```diff
@@ -0,0 +1,79 @@
+
+// Вправа №5
+const mas = [['foo', 1], ['baz', 2], ['bar', 3]];
+const ob = {};
```
Старайтеся не залишати в коді те, що ніде не використовуєте. Це допоможе підтримувати чистоту коду

Uraniadz committed:
Modul_6 Вправа 7

Uraniadz committed:
Вправа 7

Нарешті вийшло зробити вправу 7.

Uraniadz committed:
Вправа 7

Получилося зробити вправу 7

Uraniadz committed:
Вправа 7

Получилося зробити вправи з модулю 7
